client/packages/system/src/Manage/Campaigns/ListView/CampaignEditModal.tsx

```diff
@@ -0,0 +1,100 @@
+import React, { FC } from 'react';
+import {
+  useTranslation,
+  DetailContainer,
+  Box,
+  useDialog,
+  DialogButton,
+  InputWithLabelRow,
+  BasicTextInput,
+  DateUtils,
+  DateTimePickerInput,
+} from '@openmsupply-client/common';
+import { DraftCampaign, defaultDraftCampaign } from '../api';
+
+interface CampaignEditModalProps {
+  campaign: DraftCampaign;
+  isOpen: boolean;
+  onClose: () => void;
+  updateDraft: (campaign: Partial<DraftCampaign>) => void;
+  upsert: () => Promise<void>;
+}
+
+export const CampaignEditModal: FC<CampaignEditModalProps> = ({
+  campaign,
+  isOpen,
+  onClose,
+  updateDraft,
+  upsert,
+}) => {
+  const t = useTranslation();
+
+  const { Modal } = useDialog({ isOpen, onClose, disableBackdrop: true });
+
+  const { name, startDate, endDate, id } = campaign;
+
+  return (
+    <Modal
+      title={id ? 'Edit Campaign' : 'Create Campaign'}
+      cancelButton={
+        <DialogButton
+          variant="cancel"
+          onClick={() => {
+            onClose();
+            updateDraft(defaultDraftCampaign);
+          }}
+        />
+      }
+      okButton={
+        <DialogButton
+          variant="ok"
+          onClick={async () => {
+            await upsert();
+            onClose();
+            updateDraft(defaultDraftCampaign);
+          }}
+        />
+      }
+      height={1000}
```

Comment:
nit again but could make this smaller as full screen height feels unnecessary to me:

![Screenshot 2025-05-22 at 2 15 15 PM](https://github.com/user-attachments/assets/e0120559-2b8f-4b5f-9d07-543785699631)

Reply:
100%, this was leftover from the place I copy-pasted it from 😉

Looks much better now: 

![Screenshot 2025-05-23 at 2 27 02 PM](https://github.com/user-attachments/assets/7809c912-6f4b-4692-b45a-991a8f39608e)

Good catch 👍